dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/net/RequestExecutionContext.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.api.net
+
+/**
+ * Provides information about the request execution context such as the number of attempts made to
+ * execute the request in case of a retry or the response code of the previous request failure code.
+ * @param attempts the number of attempts made to execute this particular request in case of a retry.
+ * @param previousResponseCode the response code of the previous request failure code in case of a retry.
+ */
+data class RequestExecutionContext(
+    val attempts: Int = 0,
```

Comment:
> [!WARNING]
> Is this the number of attempts including the current one being made, or excluding it? It would be better to make this info explicit in the doc and in the name.